File: rust/ex2/src/main.rs
/*
 * Yes! Rust doesn't allow you to change variables by default,
 * you must specify that you are going to change a variable 
 * when you initialize it!
 *
 * Now let's try structs... Structs are like classes in a 
 * language like C++ or Java (but much simpler...). In the
 * next example we ask you to implement a method for the struct
 * Person that prints "Hi! My name is <person's name>!". This 
 * method should be called meet() and be declared similar to the
 * already implemented method age().
 *
 * There's also a sneaky bug in the main() function... see if the
 * compiler helps you with that...
 * */

struct Person {
    name: String,
    age: u64
}

impl Person {
    pub fn age(&self) {
        println!("I am {} years old!", self.age);
    }
    pub fn meet(&self) {
        //implement
    }
}


fn main() {
    let alex = Person { name: "Alex", age: 17 };
    alex.meet();
    alex.age();
}

File: go/ex2/main.go
/*
 * Yes! In Go, ":=" always declares at least one new variable, so
 * re-declaring "b" with ":=" a second time in the same scope is a
 * compile error. Using "b = 2" (a plain assignment) fixes it.
 *
 * Now let's try structs... Structs are like classes in a language
 * like C++ or Java (but much simpler...). In the next example we
 * ask you to implement a method for the struct Person that prints
 * "Hi! My name is <person's name>!". This method should be called
 * Meet() and be declared similar to the already implemented method
 * Greet().
 *
 * There's also a sneaky bug in the main() function... see if the
 * compiler helps you with that...
 * */

package main

import "fmt"

type Person struct {
	Name string
	Age  uint64
}

func (p Person) Greet() {
	fmt.Printf("I am %d years old!\n", p.Age)
}

func (p Person) Meet() {
	// implement: print "Hi! My name is <p.Name>!"
}

func main() {
	alex := Person{Name: "Alex", Age: "17"}
	alex.Meet()
	alex.Greet()
}
